### PR TITLE
first api function done

### PR DESCRIPTION
**Pull request by joshymjose**

### CODE REVIEW

File: src/common/wui_api.c [hunk 1]
```diff
@@ -7,5 +7,104 @@
...
-void get_eth_params(wui_eth_params_s eth_params) {
+#include "ini_handler.h"
+#include "eeprom.h"
+#include "string.h"
+
+#define LAN_MSK_TYPE      2 //EEPROM mask for user-defined settings (Switch between dhcp and static)
+#define IP4_TYPE_FLAG_POS 1
+
+#define IS_LAN_STATIC(flg) (flg & LAN_MSK_TYPE)
+#define IS_LAN_DHCP(flg)   (!IS_LAN_STATIC(flg))
```
Comment (michalrudolf) [inline src/common/wui_api.c]: You will need IS_LAN_ON(flg) and IS_LAN_OFF(flg) too

File: src/common/wui_api.c [hunk 2]
```diff
@@ -7,5 +7,104 @@
...
+
+static int ini_handler_func(void *user, const char *section, const char *name, const char *value) {
+
+    ETH_Config_t *tmp_config = (ETH_Config_t *)user;
+
+#define MATCH(s, n) strcmp(section, s) == 0 && strcmp(name, n) == 0
+
+    if (MATCH("lan_ip4", "type")) {
+        if (strncmp(value, "DHCP", 4) == 0 || strncmp(value, "dhcp", 4) == 0) {
+            tmp_config->lan_flag &= ~LAN_EEFLG_TYPE;
```
Comment (michalrudolf) [inline src/common/wui_api.c:27]: This should be readable, under some macro maybe.. (CHANGE_TO_DHCP(tmp_config->lan_flag))

File: src/common/wui_api.c [hunk 3]
```diff
@@ -7,5 +7,104 @@
...
+    } else {
+        return 0; /* unknown section/name, error */
+    }
+    return 1;
+}
+
+static ini_handler wui_init_handler = ini_handler_func;
+
+void load_ini_params(ETH_Config_t *config_t) {
+    ini_load_file(wui_init_handler, (void *)config_t);
```
Comment (michalrudolf) [inline src/common/wui_api.c]: wui_iniT_handler?

File: src/common/wui_api.c [hunk 4]
```diff
@@ -7,5 +7,104 @@
...
+            eeprom_set_var(EEVAR_LAN_FLAG, variant8_ui8(lan_flags));
+            eeprom_set_var(EEVAR_LAN_IP4_ADDR,
+                variant8_ui32(config_t->lan_ip4_addr.addr));
+            eeprom_set_var(EEVAR_LAN_IP4_MSK,
+                variant8_ui32(config_t->lan_ip4_msk.addr));
+            eeprom_set_var(EEVAR_LAN_IP4_GW,
+                variant8_ui32(config_t->lan_ip4_gw.addr));
+        } else {
+            uint8_t lan_flags = eeprom_get_var(EEVAR_LAN_FLAG).ui8;
+            lan_flags = 0xff;
```
Comment (michalrudolf) [inline src/common/wui_api.c]: This is wrong, lan flag holds two flags, Switch (ON, OFF) on pos 0 and Type (DHCP, STATIC) on pos 1, and both are 0 by default.

File: src/common/wui_api.c [hunk 5]
```diff
@@ -7,5 +7,104 @@
...
+    return 1;
+}
+
+static ini_handler wui_init_handler = ini_handler_func;
+
+void load_ini_params(ETH_Config_t *config_t) {
+    ini_load_file(wui_init_handler, (void *)config_t);
+}
+
+uint32_t eeprom_set_eth_params(ETH_Config_t *config_t) {
```
Comment (michalrudolf) [inline src/common/wui_api.c]: This function is probably not enough. There are cases that wants to set only LAN flag and that is not possible here.

File: src/wui/wui_api.h [hunk 6]
```diff
@@ -9,17 +9,45 @@
 #ifndef _WUI_API_H_
 #define _WUI_API_H_
 
+#include "lwip/netif.h"
+#include "eeprom.h"
```
Comment (michalrudolf) [inline src/wui/wui_api.h:13]: remove this, crossed dependencies and it's used only for string sizes which should be defined here and included into eeprom

**Review by michalrudolf — CHANGES_REQUESTED** [review]
I get the whole idea now.. I was thinking we need an interface like netifapi.h, but you only mean this few funcs... Please let me do the refactoring of this, otherwise we lose a lot of time by redoing something that is already done. As you said, there were a lot of changes in my last PR.. I promise that I will push it and check it with you step by step..

File: src/wui/wui_api.h (hunk 6 above)
Comment (joshymjose) [reply src/wui/wui_api.h:13]: it should be opposit. api function is not to include to EEPROM

**Review by michalrudolf — APPROVED** [review]

**Pull request merged**